pkg/util/pki/csr.go

```diff
@@ -242,6 +243,14 @@ func GenerateCSR(crt *v1.Certificate, optFuncs ...GenerateCSROption) (*x509.Cert
 		}
 	}
 
+	if len(crt.Spec.OtherNameSANs) != 0 {
+		otherNameExtraExtensions, err := buildExtraExtensionsOtherNameSANsForCertificate(crt)
```

Comment:
You'll have to marshall all SANs, otherwise go will skip them (see https://cs.opensource.google/go/go/+/refs/tags/go1.21.2:src/crypto/x509/x509.go;l=1166-1167).

Reply:
Please also add a test with a certificate that has both emails and UPNs defined.